esphome/components/micronova/micronova.h

```diff
@@ -0,0 +1,164 @@
+  STOVE_FUNCTION_STOVE_POWER = 5,
+  STOVE_FUNCTION_FAN_SPEED = 6,
+  STOVE_FUNCTION_STOVE_STATE = 7,
+  STOVE_FUNCTION_MEMORY_ADDRESS_SENSOR = 8,
+  STOVE_FUNCTION_WATER_TEMPERATURE = 9,
+  STOVE_FUNCTION_WATER_PRESSURE = 10,
+  STOVE_FUNCTION_POWER_LEVEL = 11,
+  STOVE_FUNCTION_CUSTOM = 12
+};
+
+class MicroNova;
+
+//////////////////////////////////////////////////////////////////////
+// Interface classes.
+class MicroNovaBaseListener {
+ public:
+  MicroNovaBaseListener() {}
+  MicroNovaBaseListener(MicroNova *m) { micronova_ = m; }
+  virtual void dump_config();
+
+  void set_micronova_object(MicroNova *m) { micronova_ = m; }
+
+  void set_function(MicroNovaFunctions f) { function_ = f; }
+  MicroNovaFunctions get_function() { return function_; }
+
+  void set_memory_location(uint8_t l) { memory_location_ = l; }
+  uint8_t get_memory_location() { return memory_location_; }
+
+  void set_memory_address(uint8_t a) { memory_address_ = a; }
+  uint8_t get_memory_address() { return memory_address_; }
+
+ protected:
+  MicroNova *micronova_{nullptr};
+  MicroNovaFunctions function_ = MicroNovaFunctions::STOVE_FUNCTION_VOID;
+  uint8_t memory_location_ = 0;
+  uint8_t memory_address_ = 0;
+};
+
+class MicroNovaSensorListener : public MicroNovaBaseListener {
+ public:
+  MicroNovaSensorListener() {}
+  MicroNovaSensorListener(MicroNova *m) : MicroNovaBaseListener(m) {}
+  virtual void request_value_from_stove();
+  virtual void process_value_from_stove(int value_from_stove);
+
+  void set_needs_update(bool u) { needs_update_ = u; }
+  bool get_needs_update() { return needs_update_; }
+
+ protected:
+  bool needs_update_ = false;
+};
+
+class MicroNovaNumberListener : public MicroNovaBaseListener {
+ public:
+  MicroNovaNumberListener(MicroNova *m) : MicroNovaBaseListener(m) {}
+  virtual void request_value_from_stove();
+  virtual void process_value_from_stove(int value_from_stove);
+
+  void set_needs_update(bool u) { needs_update_ = u; }
+  bool get_needs_update() { return needs_update_; }
+
+ protected:
+  bool needs_update_ = false;
+};
+
+class MicroNovaSwitchListener : public MicroNovaBaseListener {
+ public:
+  MicroNovaSwitchListener(MicroNova *m) : MicroNovaBaseListener(m) {}
+  virtual void set_stove_state(bool v);
+  virtual bool get_stove_state();
+
+ protected:
+  uint8_t memory_data_on_ = 0;
+  uint8_t memory_data_off_ = 0;
+};
+
+class MicroNovaButtonListener : public MicroNovaBaseListener {
+ public:
+  MicroNovaButtonListener(MicroNova *m) : MicroNovaBaseListener(m) {}
+
+ protected:
+  uint8_t memory_data_ = 0;
+};
+
+/////////////////////////////////////////////////////////////////////
+// Main component class
+class MicroNova : public PollingComponent, public uart::UARTDevice {
+ public:
+  MicroNova() {}
+
+  void setup() override;
+  void loop() override;
+  void update() override;
+  void dump_config() override;
+  void register_micronova_listener(MicroNovaSensorListener *l) { micronova_listeners_.push_back(l); }
```

Comment:
We like to always prefix member variables with `this->` so it is always clear

Please make sure to update it everywhere.
```suggestion
  void register_micronova_listener(MicroNovaSensorListener *l) { this->micronova_listeners_.push_back(l); }
```

Reply:
Done, I hope I've spotted them all.